api/types.proto

```diff
@@ -713,3 +713,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
```

Comment:
To continue a collapsed discussion about what other types there should be, as @stevvooe suggested:

Aside from container secrets, it might be good to include cluster-specific secrets, such as the swarm CA key, so it doesn't have to get written to disk.  Alternately I guess that can live in the cluster, since we probably need to encrypt the raft store at rest once it includes secrets.

What kind of secret does the registry auth one count as?  Are these cluster-wide?  Node-specific?